flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/observer/BaseObserver.java

```diff
@@ -45,7 +45,7 @@
 import java.util.Optional;
 
 /** The base observer. */
-public abstract class BaseObserver implements Observer {
+public abstract class BaseObserver implements Observer<FlinkDeployment> {
```

Comment:
We could rename this BaseDeploymentObserver/AbstractDeploymentObserver and move deployment related stuff under a deployment package , similarly we could have a `job` package for sessionjobs

Reply:
Agree, I will do it

Reply:
Two more question here:

- should we also move the `JobManagerDeploymentStatus` under the crd, since it is part of the crd.
- should we also rename the old `JobObserver` and `JobReconciler` to `ApplicationObserver` and `ApplicationReconciler` I prefer the latter

Reply:
1. Yes, lets move it to the `status` package
2. I think it should be called `ApplicationReconciler`/`ApplicationObserver` as it observes/reconciles both job and deployment.